### PR TITLE
ICU-22479 Add a new fuzzer to test more Locale methods

### PR DESCRIPTION
**Pull request by FrankYFTang**
##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22479
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [X] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by FrankYFTang** [comment]
PTAL

File: icu4c/source/test/fuzzer/locale_morph_fuzzer.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+#include <vector>
+
+#include "unicode/locid.h"
+#include "unicode/localpointer.h"
+#include "unicode/stringpiece.h"
+
+#include "locale_util.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  if (size < 1) return 0;
```
Comment (markusicu) [inline icu4c/source/test/fuzzer/locale_morph_fuzzer.cpp:21]: Please use 4 spaces of indentation per level, not 2.

**Review by markusicu — COMMENTED** [review]
mostly lgtm; please fix the indentation and then squash

**Comment by FrankYFTang** [comment]
PTAL

**Review by markusicu — COMMENTED** [review]
lgtm please squash

**Review by markusicu — APPROVED** [review]

**Pull request merged**